doc/release/compatibility.rst

```diff
@@ -0,0 +1,150 @@
+
+..  _cg_replication_protocol:
+
+Replication protocol
+--------------------
+
+An instance run on a newer release may work as:
+
+*   upstream (master) of an instance with an older release
+
+*   downstream (replica) without a database schema upgrade.
+
+The database schema upgrade (``box.schema.upgrade()``) must be performed when all replicaset instances
+run on the same Tarantool version.
+The upgrade does not cause downtime if the application does not lean on internal schema representation.
+
+..  _cg_lua_code:
+
+Lua code
+--------
+
+If a code is run on an older release, it will operate with the same effect on a
+newer one. However, only meaningful code counts.
+If any code throws an error but starts doing something useful, the change is considered compatible.
+
+A room for new functionality is still here: adding new options, more
+fields to a returning table, and more returning values (multireturn).
+
+Adding a new built-in module or a new global value is considered as the compatible change.
+
+Adding a new field to an existing metatable is okay if it is not listed in the Lua 5.1 Reference Manual. Otherwise, it should be proven that it may not break any meaningful code.
+
+Examples of compatible changes:
+
+*   Add ``__pairs``, ``__ipairs`` to a metatable of a userdata.
+    It is not from Lua 5.1, and the userdata has no default behaviour for ``pairs()`` and `ipairs()` calls.
+
+*   Add ``__lt``, ``__le`` metamethod
+    (if the attempt to use ``<``, ``<=`` and so on leads to an error before the change).
+
+Examples of NOT compatible changes:
+
+*   Add ``__pairs``, ``__ipairs`` to a metatable of a table or cdata
+    (it already has its own behavior before the change).
+
+*   Add ``__eq`` metamethod (``==`` already returns some result`).
+
+
+..  _cg_sql_code:
+
+SQL code
+--------
+
+If any request is run with the same effect, the change is
+compatible (except the requests that always lead to an error).
+
+Examples of compatible changes:
+
+*   Add a new keyword.
+*   Add a new type.
+*   Add a new built-in function.
+*   Add a new system table that starts from underscore.
+*   Add a new collation.
+*   Add an implicit or explicit cast rule for a set of operations {X} and a list
+    of types [Y] if [operation from {X}]([list of values of [Y] types]) had no
+    meaning before the change.
+
+Technically, those changes may break a working code in case of a name clash,
+but the probability of it is negligible.
+
+Examples of NOT compatible changes:
+
+*   Change how data is stored in the database.
+*   Change the result of working implicit or explicit cast.
+*   Change of a literal type.
+
+..  _cg_c_code:
+
+C code
+------
+
+If a module or a C stored procedure is run on an older release,
+it will operate with the same effect on a newer one.
+
+It is okay to add a new function or structure to the public C API.
+It must use one of the Tarantool prefixes (``box_``, ``fiber_``, ``luaT_``, ``luaM_`` and so on) or introduce a new one.
```

Comment:
```suggestion
It must use one of the Tarantool prefixes (``box_``, ``fiber_``, ``luaT_``, ``luaM_`` and so on) or introduce a new one.
```
Do I understand it correctly that the newly added function or structure must _itself_ introduce a new Tarantool prefix if it doesn't use any of the existing Tarantool prefixes?